salt/states/icinga2.py

```diff
@@ -0,0 +1,276 @@
+# -*- coding: utf-8 -*-
+'''
+Icinga2 state
+==========================
+
+.. versionadded:: TODO
+
+:depends:   - Icinga2 Python module
+:configuration: See :py:mod:`salt.modules.icinga2` for setup instructions.
+
+The icinga2 module is used to execute commands.
+Its output may be stored in a file or in a grain.
+
+.. code-block:: yaml
+
+    command_id:
+      icinga2.generate_ticket
+        - name: domain.tld
+        - output:  "/tmp/query_id.txt"
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import os.path
+
+# Import Salt libs
+import salt.utils
+
+
+def __virtual__():
+    '''
+    Only load if the icinga2 module is available in __salt__
+    '''
+    return 'icinga2.generate_ticket' in __salt__
+
+
+def generate_ticket(name, output=None, grain=None, key=None, overwrite=True):
```

Comment:
The ticket is generated on icinga2 server and used on icinga2 agent. So if that ticket changes it should always override the grain/file in order to reflect the new ticket. Not sure if this is the correct way of doing it though.